fili-sql/src/main/java/com/yahoo/bard/webservice/sql/SqlAggregationQuery.java

```diff
@@ -0,0 +1,156 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.sql;
+
+import com.yahoo.bard.webservice.data.dimension.Dimension;
+import com.yahoo.bard.webservice.druid.model.DefaultQueryType;
+import com.yahoo.bard.webservice.druid.model.aggregation.Aggregation;
+import com.yahoo.bard.webservice.druid.model.datasource.DataSource;
+import com.yahoo.bard.webservice.druid.model.datasource.QueryDataSource;
+import com.yahoo.bard.webservice.druid.model.filter.Filter;
+import com.yahoo.bard.webservice.druid.model.having.Having;
+import com.yahoo.bard.webservice.druid.model.orderby.LimitSpec;
+import com.yahoo.bard.webservice.druid.model.postaggregation.PostAggregation;
+import com.yahoo.bard.webservice.druid.model.query.AbstractDruidAggregationQuery;
+import com.yahoo.bard.webservice.druid.model.query.DruidAggregationQuery;
+import com.yahoo.bard.webservice.druid.model.query.DruidFactQuery;
+import com.yahoo.bard.webservice.druid.model.query.Granularity;
+import com.yahoo.bard.webservice.druid.model.query.QueryContext;
+
+import org.joda.time.Interval;
+
+import java.util.Collection;
+
+/**
+ * Wrapper around an {@link DruidAggregationQuery} which always reports
+ * itself as a {@link DefaultQueryType#GROUP_BY}.
+ */
+public class SqlAggregationQuery extends AbstractDruidAggregationQuery<SqlAggregationQuery> {
+
+    /**
+     * Wraps a query as a GroupBy Query.
+     *
+     * @param query  The query to wrap.
+     */
+    public SqlAggregationQuery(DruidAggregationQuery<?> query) {
+        this(
+                query.getDataSource(),
+                query.getGranularity(),
+                query.getDimensions(),
+                query.getFilter(),
+                query.getAggregations(),
+                query.getPostAggregations(),
+                query.getIntervals(),
+                query.getContext(),
+                false
+        );
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param dataSource  The datasource
+     * @param granularity  The granularity
+     * @param dimensions  The dimensions
+     * @param filter  The filter
+     * @param aggregations  The aggregations
+     * @param postAggregations  The post-aggregations
+     * @param intervals  The intervals
+     * @param context  The context
+     * @param doFork  true to fork a new context and bump up the query id, or false to create an exact copy of the
+     * context.
+     */
+    private SqlAggregationQuery(
+            DataSource dataSource,
+            Granularity granularity,
+            Collection<Dimension> dimensions,
+            Filter filter,
+            Collection<Aggregation> aggregations,
+            Collection<PostAggregation> postAggregations,
+            Collection<Interval> intervals,
+            QueryContext context,
+            boolean doFork
```

Comment:
This boolean flag is used as a part of query splitting. I'm pretty sure we're not supporting query splitting for SQL backends right now, so we probably shouldn't include this constructor at all.